### PR TITLE
Add: implement grabber_cooldown

### LINKED ISSUE #131: Not about to pause if image exist
If image file exist the "rest =" will not be kick in and then there will be no pause between each episode download, IP ban by manhuagui.com after 70~80 episode

### PR DESCRIPTION
**Pull request by eight04**
Fixes #131.
Fixes #132.

@DummySYSTEM Could you test if this branch works? You can install this branch with:
```
pip install https://github.com/eight04/ComicCrawler/archive/dev-grabber-cooldown.zip
```

### CODE REVIEW

**Comment by DummySYSTEM** [comment]
seems working well , I test it with a fresh copy

I found a better way to overcome the limit, basically I add some random time to the reset_analyze to pretent it is a human, works well with seemh, and I can lower rest_analyze to 1 instead of 3, tested on v2018.6.21 and v2018.7.18 for 2 weeks already, lower to 0.5 will be ban sometime, but rare happen

I had more mod for other code so I just put part of them in here

in batch_analyzer.py
`def get_cooldown(self, mission):
		if not hasattr(mission.module, "rest_analyze"):
			return 0
		pre_ts = self.cooldown.get(mission.module.name)
		if pre_ts is None:
			return 0
		cooldown = mission.module.rest_analyze - (time() - pre_ts) + random.uniform(0,1)
		print("pause rest_analyze",cooldown)
		return cooldown if cooldown > 0 else 0
`

**Comment by eight04** [comment]
You have to compare two approaches equally. The range should be `[-0.5, 0.5]` so you can see if it performs better with/without the random delay.

**Comment by eight04** [comment]
Also, try to implement the random delay on this branch. I will drop `rest_analyze` constant in seemh since this branch overcomes it.

**Comment by eight04** [comment]
> you mean apply [-0.5, 0.5] to rest constant or?

```python
cooldown = mission.module.rest_analyze - (time() - pre_ts) + random.uniform(0,1)
```
Adding `random.uniform(0,1)` (a random number in range `[0, 1]`) would increase the average delay.

File: comiccrawler/grabber.py [hunk 1]
```diff
@@ -20,6 +22,30 @@
...
+		if last_defer:
+			last_defer.get()
+		yield
+	except Exception as err:
+		defer.reject(err)
+		raise
+	finally:
+		@async_
+		def _():
+			time.sleep(cooldown.get(domain, 0))
```
Comment (eight04) [inline comiccrawler/grabber.py]: @DummySYSTEM You may want to change this line into
```python
delay = cooldown.get(domain, 0)
if delay != 0:
  delay += random.uniform(-0.5, 0.5)
time.sleep(delay)
```

Comment (DummySYSTEM) [reply comiccrawler/grabber.py]: get your point now

**Comment by eight04** [comment]
I'm going to merge this.

**Pull request merged**